app/services/kml/importer.rb

```
@@ -0,0 +1,234 @@
+    end
+
+    data
+  rescue StandardError => e
+    Rails.logger.warn("Failed to extract extended data: #{e.message}")
+    {}
+  end
+
+  def bulk_insert_points(batch)
+    unique_batch = batch.uniq { |record| [record[:lonlat], record[:timestamp], record[:user_id]] }
+
+    # rubocop:disable Rails/SkipsModelValidations
```

RuboCop (1.81.7)

[warning] Unnecessary disabling of `Rails/SkipsModelValidations`.

(Lint/RedundantCopDisableDirective)